lib/bitcoin/lock/LockMonitor.ts

```diff
@@ -0,0 +1,304 @@
+import BitcoinClient from '../BitcoinClient';
+import BitcoinError from '../BitcoinError';
+import BitcoinLockTransactionModel from '../models/BitcoinLockTransactionModel';
+import ErrorCode from '../ErrorCode';
+import LockIdentifier from '../models/LockIdentifierModel';
+import LockIdentifierSerializer from './LockIdentifierSerializer';
+import LockResolver from './LockResolver';
+import LockTransactionModel from './../models/LockTransactionModel';
+import LockTransactionType from './../enums/LockTransactionType';
+import MongoDbLockTransactionStore from './MongoDbLockTransactionStore';
+import ValueTimeLockModel from './../../common/models/ValueTimeLockModel';
+
+/**
+ * Structure (internal to this class) to track the information about lock.
+ */
+interface LockInformation {
+  currentValueTimeLock: ValueTimeLockModel | undefined;
+  latestSavedLockInfo: LockTransactionModel | undefined;
+
+  // Need to add a 'state'
+}
+
+/**
+ * Encapsulates functionality to monitor and create/remove amount locks on bitcoin.
+ */
+export default class LockMonitor {
+
+  private periodicPollTimeoutId: number | undefined;
+
+  private currentValueTimeLock: LockInformation | undefined;
+
+  private lockResolver: LockResolver;
+
+  constructor (
+    private bitcoinClient: BitcoinClient,
+    private lockTransactionStore: MongoDbLockTransactionStore,
+    private pollPeriodInSeconds: number,
+    private desiredLockAmountInSatoshis: number,
+    private lockPeriodInBlocks: number,
+    private firstLockFeeAmountInSatoshis: number) {

```

Comment:
Seems to imply that this amount will be gone to the minor in one shot in the first lock. Perhaps:
```suggestion
    private satoshisToPayForLocks: number) {

```

Reply:
Changing it to ````transactionFeesAmountInSatoshis````.